uplink/storage/streams/store.go

```diff
@@ -418,42 +413,38 @@ func (s *streamStore) Delete(ctx context.Context, path Path, pathCipher storj.Ci
 		return err
 	}
 
-	lastSegmentPath, err := createSegmentPath(ctx, -1, path.Bucket(), encPath)
-	if err != nil {
-		return err
-	}
+	// TODO maybe for now just use GetObject and decrypt num of segments from metainfo
+	// to avoid two satellite requestes ??
 
-	lastSegmentMeta, err := s.segments.Meta(ctx, lastSegmentPath)
+	// TODO do it in batch
+	streamID, err := s.metainfo.BeginDeleteObject(ctx, metainfo.BeginDeleteObjectParams{
+		Bucket:        []byte(path.Bucket()),
+		EncryptedPath: []byte(encPath.Raw()),
+	})
 	if err != nil {
 		return err
 	}
 
-	streamInfo, streamMeta, err := TypedDecryptStreamInfo(ctx, lastSegmentMeta.Data, path, s.encStore)
+	// TODO handle `more`
+	items, _, err := s.metainfo.ListSegmentsNew(ctx, metainfo.ListSegmentsParams{
```

Comment:
I think we need to handle this, otherwise we'll start leaking data after 6.4GB sized uploads, but I'm fine with doing a separate PR for this.

Reply:
I will prepare PR for that when this will be merged.